packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -55,8 +199,132 @@ export class EcsDeploymentGroup {
     return new ImportedEcsDeploymentGroup(scope, id, attrs);
   }
 
-  private constructor() {
-    // nothing to do until CFN supports ECS deployment groups
+  public readonly application: IEcsApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+  public readonly deploymentConfig: IEcsDeploymentConfig;
+  /**
+   * The role for this CodeDeploy to use when deploying to this deployment group.
+   */
+  public readonly role: iam.IRole;
+  private readonly alarms: cloudwatch.IAlarm[];
+
+  constructor(scope: Construct, id: string, props: EcsDeploymentGroupProps) {
+    super(scope, id, {
+      physicalName: props.deploymentGroupName,
+    });
+
+    this.application = props.application || new EcsApplication(this, 'Application');
+    this.alarms = props.alarms || [];
+
+    this.role = props.role || new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com'),
+    });
+    this.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSCodeDeployRoleForECS'));
+    this.deploymentConfig = props.deploymentConfig || EcsDeploymentConfig.ALL_AT_ONCE;
+
+    // Change ECS Service deployment controller to codedeploy
+    for (const service of props.services) {
+      const cfnService = service.node.defaultChild as ecs.CfnService;
+      cfnService.deploymentController = { type: ecs.DeploymentControllerType.CODE_DEPLOY };
+
+      // Remove revision from ECS Service task definition to allow Blue/Green deploys to work
+      cfnService.taskDefinition = service.taskDefinition.family;
+      service.node.addDependency(service.taskDefinition);
```

Comment:
Can you explain more?  I'm not following why this is necessary

Reply:
This is how I avoid CloudFormation trying to update the ECS service when the TaskDefinition revision changes. By removing revision and only using family, it is only deployed on creation and ignored going forward.